HWP conversion fails on MacOS M1

HWP conversion was recently added in https://github.com/freedomofpress/dangerzone/pull/460 and was tested across various platforms (Windows, MacOS, Linux) through our CI. However, when running our tests on a MacOS M1 platform (which is not available through any of our CI runners), we get the following error:

```
[DEBUG] Marking doc UWYs15 as 'converting'
[INFO ] > /usr/local/bin/docker run --network none -u dangerzone --security-opt=no-new-privileges:true --cap-drop all --rm -v /var/folders/7c/49675rcx2nb0lwjms_6wmdpr0000gs/T/tmp6rzrqzt1/unsafe/input_file:/tmp/input_file:Z -v /var/folders
/7c/49675rcx2nb0lwjms_6wmdpr0000gs/T/tmp6rzrqzt1/pixels:/tmp/dangerzone:Z -e ENABLE_TIMEOUTS=1 dangerzone.rocks/dangerzone /usr/bin/python3 -m dangerzone.conversion.doc_to_pixels
[INFO ] [doc UWYs15] 0% UNTRUSTED> Installing LibreOffice extension 'h2orestart.oxt'
[INFO ] [doc UWYs15] 0% UNTRUSTED> Converting to PDF using LibreOffice                                 
[INFO ] [doc UWYs15] 3% UNTRUSTED> Calculating number of pages
[ERROR] [doc UWYs15] 3% UNTRUSTED> PDF file is corrupted
```

_(taken from the `dangerzone-cli`)_

If one tries to convert the file to PDF via Libreoffice:

```
libreoffice --headless --infilter="Hwp2002_Reader" --convert-to pdf:writer_pdf_Export <file>
```
they will get the following generic error:

```
Error: source file could not be loaded
```

Note that the exact same operation in an Intel MacOS device works.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,10 @@ since 0.4.1, and this project adheres to [Semantic Versioning](https://semver.or
 
 ## Unreleased
 
+### Changed
+
+- Feature: Add support for HWP/HWPX files (Hancom Office) for macOS Apple Silicon devices ([issue #498](https://github.com/freedomofpress/dangerzone/issues/498), thanks to [@OctopusET](https://github.com/OctopusET))
+
 ## Dangerzone 0.5.1
 
 ### Fixed

--- a/README.md
+++ b/README.md
@@ -53,8 +53,7 @@ Dangerzone can convert these types of document into safe PDFs:
 - ODF Graphics (`.odg`)
 - Hancom HWP (Hangul Word Processor) (`.hwp`, `.hwpx`)
   * Not supported on
-    [MacOS with Apple Silicon CPU](https://github.com/freedomofpress/dangerzone/issues/498)
-    or [Qubes OS](https://github.com/freedomofpress/dangerzone/issues/494)
+    [Qubes OS](https://github.com/freedomofpress/dangerzone/issues/494)
 - Jpeg (`.jpg`, `.jpeg`)
 - GIF (`.gif`)
 - PNG (`.png`)

--- a/dangerzone/conversion/doc_to_pixels.py
+++ b/dangerzone/conversion/doc_to_pixels.py
@@ -10,7 +10,6 @@ Here are the steps, with progress bar percentages:
 import asyncio
 import glob
 import os
-import platform
 import re
 import shutil
 import sys
@@ -189,13 +188,6 @@ class DocumentToPixels(DangerzoneConverter):
             #
             #     https://github.com/freedomofpress/dangerzone/issues/494
             #     https://github.com/freedomofpress/dangerzone/issues/498
-            if libreoffice_ext == "h2orestart.oxt" and platform.machine() in (
-                "arm64",
-                "aarch64",
-            ):
-                raise ValueError(
-                    "HWP / HWPX formats are not supported in ARM architectures"
-                )
             if libreoffice_ext == "h2orestart.oxt" and running_on_qubes():
                 raise errors.DocFormatUnsupportedHWPQubes()
             if libreoffice_ext:

--- a/dangerzone/gui/main_window.py
+++ b/dangerzone/gui/main_window.py
@@ -553,15 +553,13 @@ class DocSelectionWidget(QtWidgets.QWidget):
         self.file_dialog.setWindowTitle("Open Documents")
         self.file_dialog.setFileMode(QtWidgets.QFileDialog.ExistingFiles)
 
-        # XXX: We disable loading HWP/HWPX files on Qubes or MacOS M1 platforms, because
-        # H2ORestart does not work there. See:
+        # XXX: We disable loading HWP/HWPX files on Qubes, because H2ORestart does not work there.
+        # See:
         #
         # https://github.com/freedomofpress/dangerzone/issues/494
-        # https://github.com/freedomofpress/dangerzone/issues/498
         hwp_filters = "*.hwp *.hwpx"
-        if platform.machine() in ("arm64", "aarch64") or is_qubes_native_conversion():
+        if is_qubes_native_conversion():
             hwp_filters = ""
-
         self.file_dialog.setNameFilters(
             [
                 f"Documents (*.pdf *.docx *.doc *.docm *.xlsx *.xls *.pptx *.ppt *.odt *.odg *.odp *.ods {hwp_filters} *.jpg *.jpeg *.gif *.png *.tif *.tiff)"

--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -4,7 +4,6 @@ import base64
 import contextlib
 import copy
 import os
-import platform
 import re
 import shutil
 import sys
@@ -308,7 +307,7 @@ class TestCliConversion(TestCliBasic):
 class TestExtraFormats(TestCli):
     @for_each_external_doc("*hwp*")
     def test_hancom_office(self, doc: str) -> None:
-        if platform.machine() in ("arm64", "aarch64") or is_qubes_native_conversion():
+        if is_qubes_native_conversion():
             pytest.skip("HWP / HWPX formats are not supported on this platform")
         with tempfile.NamedTemporaryFile("wb", delete=False) as decoded_doc:
             with open(doc, "rb") as encoded_doc:

